### ISSUE TITLE
Noisy / grainy image with ED097TC2

### ISSUE THREAD

**Issue by 2dom** [vroland/epdiy#52]
I just received my ED097TC2 panel to replace the broken ED097OC4. While it appears to be of better contrast it is also a lot more "grainy" (see right side of attached picture). Any suggestions what I can tweak to get rid of this?

Fun fact - if I place my hand on the display during update (placed it o the left side) the grain appears to be much reduced. 

![IMG_20210106_123041](https://user-images.githubusercontent.com/4075404/103764316-858ecf00-501b-11eb-85b9-0655804c10f8.jpg)

**Comment by vroland** [comment]
You're right, when comparing the displays directly the TC2 seems noisier (though maybe not as much?) for me as well. I don't seem to have your magical hands though :/ Can you localize where your hand has an effect? I don't have an oscilloscope myself, but if I know what to look for I may be able to borrow one.
Is it the same if you first darken the display and then draw the image in white (The grayscale test example does both)?

**Comment by 2dom** [comment]
Will try white on black. I presume that the hand has some electrostatic effect. Maybe changes the off-set or something. I noticed this before - on another display I had put sum electrical tape on the back and it was clearly visible in the image.

**Comment by vroland** [comment]
Hm, in my e-paper laptop I have the whole controller board taped to the back of the display with electrical tape but it's not visible in any way. Maybe because it is shielded? 
Which revision of the board are you using, btw?

**Comment by mcer12** [comment]
@2dom just to be sure, did you adjust the voltages? According to datasheet it requires +27V (instead of usual +22V). And did you adjust VCOM according to the sticker?

**Comment by 2dom** [comment]
I did adapt vcom but not to 27V ... Will have to give this a go ☺️

**Comment by 2dom** [comment]
Thanks for the reminder - did not actually read the datasheet

**Comment by 2dom** [comment]
@vroland Checking the Revision I have to postpone until I take the picture frame apart next time

**Comment by 2dom** [comment]
@mcer12 ... Can you send me the datasheet please?

**Comment by mcer12** [comment]
@2dom https://www.welectron.com/mediafiles/datasheets/waveshare/Waveshare-9.7inch-e-Paper-HAT-Datasheet.PDF

**Comment by 2dom** [comment]
Great - thanks!

**Comment by vroland** [comment]
@2dom Did you already check if the voltage has any effect?

**Comment by 2dom** [comment]
No difference unfortunately

**Comment by mcer12** [comment]
@2dom what version of the board are you using?

**Comment by 2dom** [comment]
Version 2On 19 Jan 2021 16:18, mcer12 <notifications@github.com> wrote:
@2dom what version of the board are you using?

**Comment by mcer12** [comment]
@2dom that board uses linear regulators for +-15V rails as far as I know. And I also think the booster is driven from 3.3V which si pretty bad so maybe it's having trouble driving the large display. The V4 uses powerful booster for +-15V. @vroland can probably elaborate on that.

**Comment by vroland** [comment]
Could be a reason, But I noticed one of my TC2s being grainy with the newer versions as well. 
@2dom Does it make a difference if you make the display black first and then draw in white vs directly drawing in black on a white display?

**Comment by 2dom** [comment]
I tried both...pretty much the same effect

**Comment by vroland** [comment]
Hm, not sure what else to try. Maybe in the long therm we could reverse engineer the original waveform files. Unfortunately, I can't really find any information about the file format.

**Comment by mcer12** [comment]
@2dom @vroland I bought a second ED097TC2 and the new one is grainy, the old one isn't. So it seems like it's display-speific or some displays are more susceptible than others. I will investigate a bit further on this but I can't think of anything particular that could cause this on one display and not the other. Maybe it's just cheap low quality panels that were stolen from factory garbage, who knows.
